### PR TITLE
chore: add schema and field mappings for SyncConfig and Sync

### PR DESCRIPTION
**Pull request by albertyfwu**
## Test Plan

[Describe test plan here]

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: packages/core/mappers/sync_config.ts [hunk 1]
```diff
@@ -21,7 +21,15 @@ const fromSyncConfigDataModel = (config: Prisma.JsonValue): SyncConfigData => {
   if (!config || typeof config !== 'object' || Array.isArray(config)) {
     throw new Error('SyncConfig config is missing');
   }
-  return config as unknown as SyncConfigData;
+  const configToReturn = config as unknown as SyncConfigData;
+  return {
+    ...configToReturn,
+    rawObjects: configToReturn.rawObjects?.map((rawObject) => ({
+      ...rawObject,
+      // TODO: do a proper migration later to ensure that schema is always set
```
Comment (asdfryan) [inline packages/core/mappers/sync_config.ts:29]: Can you write a backfill endpoint?

File: packages/types/sync_config.ts [hunk 2]
```diff
@@ -27,21 +27,39 @@ export type SyncConfigData = {
 export type SyncStrategyConfig = {
   periodMs: number;
   strategy: SyncType;
+  // When this is undefined, we treat it as "true" to be backwards compatible.
+  // TODO: When we have time, migrate this to be a required field.
+  enableSyncOnConnectionCreation?: boolean;
```
Comment (albertyfwu) [inline packages/types/sync_config.ts:32]: i could make this `doNotEnableSyncOnConnectionCreation` to make it more naturally backwards compatible. thoughts?

File: packages/types/sync.ts [hunk 3]
```diff
@@ -8,6 +8,18 @@ type BaseSync = {
   syncConfigId?: string;
   forceSyncFlag: boolean; // flag: whether to transition a sync to the phase "created"
   version: 'v1' | 'v2';
+  // If this is undefined, we treat it as "true" to be backwards compatible.
+  // TODO: This should be required
+  enabled?: boolean;
```
Comment (albertyfwu) [inline packages/types/sync.ts:13]: i could make this `disabled` to make it more naturally backwards compatible. thoughts?

Comment (asdfryan) [reply packages/types/sync.ts:13]: I like `paused` better

File: packages/types/sync_config.ts (hunk 2 above)
Comment (asdfryan) [reply packages/types/sync_config.ts:32]: I like `startSyncOnConnection`

File: packages/core/mappers/sync_config.ts (hunk 1 above)
Comment (albertyfwu) [reply packages/core/mappers/sync_config.ts]: outdated comment

File: packages/types/sync_config.ts (hunk 2 above)
Comment (albertyfwu) [reply packages/types/sync_config.ts:32]: i was thinking of doing that originally, but the property on `sync` is `enabled`. isn't that a bit inconsistent?

Comment (albertyfwu) [reply packages/types/sync_config.ts:32]: ok sure

File: packages/types/sync.ts (hunk 3 above)
Comment (albertyfwu) [reply packages/types/sync.ts:13]: ok

File: packages/types/sync_config.ts [hunk 4]
```diff
@@ -27,21 +27,39 @@ export type SyncConfigData = {
 export type SyncStrategyConfig = {
   periodMs: number;
   strategy: SyncType;
+  // When this is undefined, we treat it as "true" to be backwards compatible.
+  // TODO: When we have time, migrate this to be a required field.
+  startSyncOnConnectionCreation?: boolean;
```
Comment (asdfryan) [inline packages/types/sync_config.ts:32]: We should set this explicitly on the FE

**Review by asdfryan — APPROVED** [review]

**Review by asdfryan — APPROVED** [review]

**Pull request merged**